src/components/MenuHeader/SearchDesktop.js

```diff
@@ -0,0 +1,203 @@
+import styled from 'styled-components';
+import { useContext, useState} from 'react';
+import { AiOutlineSearch } from 'react-icons/ai';
+import ShowMenuContext from '../../contexts/ShowMenuContext';
+import UserContext from "../../contexts/UserContext";
+import { Link } from "react-router-dom";
+import {DebounceInput} from 'react-debounce-input';
+import { getUsers } from '../../Service';
+
+export default function SearchDesktop(){
+
+    const {showMenu, setShowMenu} = useContext(ShowMenuContext);
+    const {user} = useContext(UserContext);
+    const [usersSearch, setUsersSearch] = useState("")
+    const [usersFound, setUsersFound] = useState("")
+
+    function toggleMenu(event){
+
+        event.stopPropagation();
+           
+        if(showMenu === true){
+            setShowMenu(!showMenu);
+          }
+    }
+
+    function search(){
+        
+        getUsers(user.token, usersSearch)
+        .then((res) => {    
+            sortUsers(res.data.users)                               
+            }) 
+        .catch((res) => {     
+        setUsersFound("")                                
+            });                  
+    } 
+
+    function sortUsers(dataUser){
+
+        const sortUsersFound = dataUser.sort((x, y) =>
+
+        (x.isFollowingLoggedUser === y.isFollowingLoggedUser) ? 0 : x.isFollowingLoggedUser ? -1 : 1)
+
+        setUsersFound(sortUsersFound)    
+
+    }
+        
+    return (
+        <>
+            <SearchandResultsStyled onClick={toggleMenu}>
+                <RelativeStyled>      
+                <DebounceInputStyled
+                minLength={3}
+                debounceTimeout={300}
+                type="text" 
+                placeholder="Search for people and friends"
+                value={usersSearch}
+                onChange={(e) => setUsersSearch(e.target.value)} 
+                onInput={search()}/>
```

Comment:
Aqui no `onInput`  não precisaria de uma arrorFunction para chamar a função?

Reply:
acho que n Bru, tá funcionando assim

Reply:
acho que teria que ser arrow se eu fosse passar algum parâmetro pra função que eu to chamando